src/shared/components/Button/Button.tsx

```diff
@@ -46,7 +46,7 @@ const ButtonComponent: React.ForwardRefRenderFunction<HTMLButtonElement, ButtonP
       size={size}
       ref={ref}
     >
-      {icon && <StyledIcon disabled={disabled} name={icon} />}
+      {icon && <StyledIcon disabled={disabled} name={icon} size={size} />}
       {children && <span>{children}</span>}
```

Comment:
There are some issues with typography. The font sizes don't match what's currently in Figma. Also the line heights are wrong producing wrong button sizes. This isn't necessarily a fault in your work here, but rather some deeper problems. I fixed some of those in my image crop dialog. I think I'll open a separate PR to make sure typography matches exactly what's in the design, hopefully that helps